### ISSUE TITLE
ListView Bindings fire multiple times on Android SDK 24+

### ISSUE THREAD

**Issue by samhouts** [xamarin/Xamarin.Forms#2080]
### Description

Migrated from https://bugzilla.xamarin.com/show_bug.cgi?id=60699

I am using Xamarin.Forms with a ListView with some bindings, targeting iOS and Android. When running the app on Android SDK 23, performance is good; however, starting on SDK 24 on physical devices (Nexus 5X is the model tested) I see some performance issues when scrolling the list. When using Visual Studio to debug, I see that the binding events are being fired haphazardly, sometimes firing twice in succession and sometimes for the all the items in the list – I have reproduced this with Xamarin.Forms 2.4.0.282. On iOS, the duplicate entries exist but only fire for the new elements being rendered.

Sample output (Android SDK 25):
```10/18/2017 1:57:01 PM [BindingFired] 10 item
10/18/2017 1:57:01 PM [BindingFired] This is an item description.
10/18/2017 1:57:01 PM [BindingFired] http://lorempixel.com/100/100/sports/
10/18/2017 1:57:01 PM [BindingFired] 11 item
10/18/2017 1:57:01 PM [BindingFired] This is an item description.
10/18/2017 1:57:03 PM [BindingFired] http://lorempixel.com/100/100/sports/
10/18/2017 1:57:03 PM [BindingFired] 5 item
10/18/2017 1:57:03 PM [BindingFired] This is an item description.
10/18/2017 1:57:04 PM [BindingFired] http://lorempixel.com/100/100/sports/
10/18/2017 1:57:04 PM [BindingFired] 6 item
10/18/2017 1:57:04 PM [BindingFired] This is an item description.
10/18/2017 1:57:05 PM [BindingFired] http://lorempixel.com/100/100/sports/
10/18/2017 1:57:05 PM [BindingFired] 12 item
10/18/2017 1:57:05 PM [BindingFired] This is an item description.
10/18/2017 1:57:05 PM [BindingFired] http://lorempixel.com/100/100/sports/
10/18/2017 1:57:05 PM [BindingFired] 12 item
10/18/2017 1:57:05 PM [BindingFired] This is an item description.
10/18/2017 1:57:05 PM [BindingFired] http://lorempixel.com/100/100/sports/
```
The ListView is defined as follows:
```<ListView x:Name="ItemsListView" ItemsSource="{Binding Items}" 
                VerticalOptions="FillAndExpand" HasUnevenRows="true" 
                RefreshCommand="{Binding LoadItemsCommand}" 
                IsPullToRefreshEnabled="true" IsRefreshing="{Binding IsBusy, Mode=OneWay}" 
                CachingStrategy="RecycleElement" ItemSelected="OnItemSelected">
```
On Android SDK 23 and below, the bindings are fired in order as they appear on the screen, never duplicate.

The performance problem is still apparent with HasUnevenRows disabled. I've captured profilers of both SDK23 and SDK25 where the issue can be seen, also updated the sample with both recommendations.

The real issue is the difference in performance between both SDKs - is there any difference in how the bindings are fired in each SDK that would explain the difference in performance?

### Basic Information

- Version with issue: 2.4.0.282
- Last known good version: unknown
- IDE: Visual Studio
- Platform Target Frameworks: 
  - Android: 24+
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices: Nexus 5X

### Reproduction Link

See Bugzilla ticket.

**Comment by samhouts** [comment]
I believe that this is related to #1908 and may possibly be resolved by #1973.

**Comment by KalyanXamarin** [comment]
Which Xamarin forms version it is fixed?

**Comment by samhouts** [comment]
@KalyanXamarin The associated PR was merged into 3.0. If you're still experiencing the problem, please open a new issue. Thanks!
